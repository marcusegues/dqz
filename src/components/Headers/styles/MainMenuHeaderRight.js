// @flow
// $FlowFixMe
import { StyleSheet } from 'react-native';

export const mainMenuHeaderRightStyle = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
});
