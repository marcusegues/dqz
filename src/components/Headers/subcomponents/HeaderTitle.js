import React from 'react';

import { Text } from 'react-native';

const HeaderTitle = ({ text }) => (
  <Text
    style={{
      fontSize: 18,
      color: '#1A1A1A',
      fontFamily: 'roboto_regular',
    }}
  >
    {text}
  </Text>
);

export default HeaderTitle;
