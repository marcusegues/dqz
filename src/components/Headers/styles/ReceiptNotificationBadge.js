import { StyleSheet } from 'react-native';
import { moderateScale, scale } from '../../../styles/Scaling';
import { MAIN_RED } from '../../../styles/colors';

export default StyleSheet.create({
  container: {
    minWidth: scale(105),
    maxWidth: scale(180),
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: MAIN_RED,
    marginRight: scale(10),
    paddingVertical: 4,
    paddingHorizontal: 5,
    borderRadius: 20,
  },
  amountText: {
    marginHorizontal: 10,
    color: 'white',
    fontFamily: 'roboto_regular',
    fontSize: moderateScale(14),
  },
  receiptIcon: {
    marginLeft: 5,
  },
});
