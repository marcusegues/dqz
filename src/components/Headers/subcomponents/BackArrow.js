import React from 'react';
import { Ionicons } from '@expo/vector-icons';
import Touchable from 'react-native-platform-touchable';
import { MAIN_RED } from '../../../styles/colors';

const BackArrow = ({ onPress }) => (
  <Touchable onPress={onPress}>
    <Ionicons name="md-arrow-back" size={32} color={MAIN_RED} />
  </Touchable>
);

export default BackArrow;
