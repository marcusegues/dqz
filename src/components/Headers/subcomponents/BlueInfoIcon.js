// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { moderateScale } from '../../../styles/Scaling';

export const BlueInfoIcon = () => (
  <Touchable onPress={() => {}}>
    <MaterialCommunityIcons
      name="information"
      size={moderateScale(30)}
      color="#006699"
    />
  </Touchable>
);
