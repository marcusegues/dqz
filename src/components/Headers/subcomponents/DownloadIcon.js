// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { MaterialIcons } from '@expo/vector-icons';
import { MAIN_RED } from '../../../styles/colors';
import { moderateScale } from '../../../styles/Scaling';
import type { Navigation } from '../../../types/generalTypes';

type DownloadIconProps = {
  navigation: Navigation,
};

export const DownloadIcon = ({ navigation }: DownloadIconProps) => (
  <Touchable
    onPress={() => {
      navigation.goBack();
    }}
  >
    <MaterialIcons name="get-app" size={moderateScale(28)} color={MAIN_RED} />
  </Touchable>
);
