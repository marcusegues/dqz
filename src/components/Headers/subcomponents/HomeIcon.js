// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import Entypo from '@expo/vector-icons/Entypo';
import { moderateScale } from '../../../styles/Scaling';
import type { Navigation } from '../../../types/generalTypes';
import { GREY } from '../../../styles/colors';

type HomeIconProps = {
  navigation: Navigation,
};

export const HomeIcon = ({ navigation }: HomeIconProps) => (
  <Touchable
    onPress={() =>
      navigation.dispatch({ type: 'NAVIGATE', screen: 'MainMenu' })
    }
  >
    <Entypo name="home" size={moderateScale(27)} color={GREY} />
  </Touchable>
);
