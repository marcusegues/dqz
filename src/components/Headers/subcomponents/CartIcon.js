// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { MaterialIcons } from '@expo/vector-icons';
import { moderateScale } from '../../../styles/Scaling';
import type { Navigation } from '../../../types/generalTypes';
import { receiptNotificationBadgeStyle } from '../styles/ReceiptNotificationBadge';
import { MAIN_RED } from '../../../styles/colors';

type CartIconProps = {
  navigation: Navigation,
};

export const CartIcon = ({ navigation }: CartIconProps) => (
  <Touchable
    onPress={() =>
      navigation.dispatch({ type: 'NAVIGATE', screen: 'QuestionAnswer' })
    }
  >
    <MaterialIcons
      name="shopping-cart"
      size={moderateScale(30)}
      color={MAIN_RED}
      style={receiptNotificationBadgeStyle.receiptIcon}
    />
  </Touchable>
);
