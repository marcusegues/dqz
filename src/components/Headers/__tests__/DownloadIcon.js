// @flow
import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { MaterialIcons } from '@expo/vector-icons';
import { DownloadIcon } from '../subcomponents/DownloadIcon';
import { MAIN_RED } from '../../../styles/colors';
import { moderateScale } from '../../../styles/Scaling';

const clickFn = jest.fn();

describe('Download Icon TestSuite', () => {
  test('DownloadIcon component renders correctly', () => {
    const tree = renderer
      // $FlowFixMe
      .create(<DownloadIcon navigation={{ state: {} }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('we can click it', () => {
    const component = shallow(
      // $FlowFixMe
      <DownloadIcon navigation={{ state: { params: { onPress: clickFn } } }} />
    );
    component.find(Touchable).simulate('press');
    expect(clickFn).toBeCalled();
    expect(component.dive()).toMatchSnapshot();
  });
  test('it contains icon', () => {
    expect.assertions(2);
    const component = shallow(
      // $FlowFixMe
      <DownloadIcon navigation={{ state: { params: { onPress: clickFn } } }} />
    );
    expect(component.find(MaterialIcons)).toHaveLength(1);
    expect(
      component.containsMatchingElement(
        <MaterialIcons
          name="get-app"
          size={moderateScale(28)}
          color={MAIN_RED}
        />
      )
    ).toBeTruthy();
  });
});
