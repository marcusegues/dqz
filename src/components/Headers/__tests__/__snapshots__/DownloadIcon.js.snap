// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Icon TestSuite CartIcon component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;

exports[`Download Icon TestSuite DownloadIcon component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;
